iphone/Classes/LocaleModule.m

```diff
@@ -84,6 +84,12 @@ - (NSNumber *)parseDecimal:(NSString *)text withLocaleId:(id)localeId
 
   // Remove localized thousands separators from string if they exist. NSScanner fails to parse them.
   NSString *thousandsSeparator = [locale objectForKey:NSLocaleGroupingSeparator];
+  // They changed France's separator character!
+  // Treat \u00a0 and \u202f interchangeably
+  // https://haacked.com/archive/2020/05/17/french-thousand-separator-mystery/
+  if ([thousandsSeparator isEqualToString:@"\u202f"]) {
+    text = [text stringByReplacingOccurrencesOfString:@"\u00a0" withString:@"\u202f"];
+  }
```

Comment:
I have an alternative solution below. If the localized thousands separator is whitespace, then I strip out all whitespace character types, include ASCII spaces. This allows it to correctly parse the **reverse** issue with a string containing `\u202F` on iOS12 which defaults to `\u00A0`.
https://github.com/appcelerator/titanium_mobile/pull/11824